### ISSUE TITLE
Add total params to weights_summary table

### ISSUE THREAD

**Issue by george-gca** [PyTorchLightning/pytorch-lightning#4222]
## 🚀 Feature
Add total number of parameters when printing the weights_summary table.

### Motivation

Since the total number of parameters for each layer is already calculated, it would be really informative if a total sum of number of parameters were also provided. Something like https://github.com/TylerYep/torch-summary or https://github.com/nmhkahn/torchsummaryX provide in their table.

### Pitch

When printing weights_summary, there should also appear a summary of the whole model information, something like:
```
                      Totals
Total params          21.84k
Trainable params      21.84k
Non-trainable params     0.0
```

### Alternatives

Maybe use one of the given tools (https://github.com/TylerYep/torch-summary or https://github.com/nmhkahn/torchsummaryX) inside?

### Additional context

Example from https://github.com/TylerYep/torch-summary model summary:
```
==========================================================================================
Layer (type:depth-idx)                   Output Shape              Param #
==========================================================================================
├─Conv2d: 1-1                            [-1, 10, 24, 24]          260
├─Conv2d: 1-2                            [-1, 20, 8, 8]            5,020
├─Dropout2d: 1-3                         [-1, 20, 8, 8]            --
├─Linear: 1-4                            [-1, 50]                  16,050
├─Linear: 1-5                            [-1, 10]                  510
==========================================================================================
Total params: 21,840
Trainable params: 21,840
Non-trainable params: 0
==========================================================================================
Input size (MB): 0.00
Forward/backward pass size (MB): 0.05
Params size (MB): 0.08
Estimated Total Size (MB): 0.14
==========================================================================================
```

Example from https://github.com/nmhkahn/torchsummaryX model summary:
```
=================================================================
                Kernel Shape     Output Shape  Params Mult-Adds
Layer
0_conv1        [1, 10, 5, 5]  [1, 10, 24, 24]   260.0    144.0k
1_conv2       [10, 20, 5, 5]    [1, 20, 8, 8]   5.02k    320.0k
2_conv2_drop               -    [1, 20, 8, 8]       -         -
3_fc1              [320, 50]          [1, 50]  16.05k     16.0k
4_fc2               [50, 10]          [1, 10]   510.0     500.0
-----------------------------------------------------------------
                      Totals
Total params          21.84k
Trainable params      21.84k
Non-trainable params     0.0
Mult-Adds             480.5k
=================================================================
```

**Comment by SkafteNicki** [comment]
Related to https://github.com/PyTorchLightning/pytorch-lightning/issues/1833
Please feel free to send a PR with the suggested enhancement!

**Comment by george-gca** [comment]
Implemented in #4521
